wasp-ai/src/server/operations.ts

```diff
@@ -222,7 +209,46 @@ export const getStats = (async (_args, context) => {
 }) satisfies GetStats<{}>;
 
 export const getNumProjects = (async (_args, context) => {
-  const { Project } = context.entities;
-  const numProjects = await Project.count();
-  return numProjects;
+  return context.entities.Project.count();
 }) satisfies GetNumProjects<{}>;
+
+export const getProjectsByUser: GetProjectsByUser<void, Project[]> = async (_args, context) => {
+  if (!context.user) {
+    throw new HttpError(401, "Not authorized");
+  }
+  return await context.entities.Project.findMany({
+    where: {
+      user: {
+        id: context.user.id,
+      },
+    },
+  });
+};
+
+import { CheckIfUserStarredWasp } from "@wasp/queries/types";
+
+export const checkIfUserStarredWasp: CheckIfUserStarredWasp<{ username: string }, boolean> = async (
+  { username },
+  context
+) => {
+  if (!context.user) {
+    throw new HttpError(401, "Not authorized");
+  }
+  let status = false;
+  try {
+    const response = await fetch(`https://api.github.com/user/starred/wasp-lang/wasp`, {
+      method: "GET",
+      headers: {
+        Authorization: "Basic " + btoa(username + ":" + process.env.GITHUB_PERSONAL_ACCESS_TOKEN),
+      },
+    });
+    console.log(response.status);
+    if (response.status === 204) {
+      status = true;
+    }
+  } catch (error) {
+    console.error("Error:", error);
+  } finally {
+    return status;
+  }
```

Comment:
So if error happens, you will return `false`, not an error.
Is that ok? Wouldn't we rather return an error in that case?
And then on frontend, if we get error, we would rather not bother them with leaving a star? Because right now, let's say we breach some rate limit or something, and we get all errors, then we will be bothering people who even starred.

Reply:
Is this comment irrelevant now?

Reply:
yes we completely removed this function due to the GH API issue